account-info/src/urls.rs

```diff
@@ -1,3 +1,44 @@
+use contract::{contract_api::{runtime, storage}, unwrap_or_revert::UnwrapOrRevert};
+use types::{Key, URef, account::AccountHash};
+use super::ContractError;
+
+pub const URLS_DICT: &str = "urls";
+
 pub struct Urls {
     dict_uref: URef
+}
+
+impl Urls {
+    pub fn new() -> Urls {
+        let dict_key: Key = runtime::get_key(URLS_DICT).unwrap_or_revert();
+        let dict_uref: &URef = dict_key.as_uref().unwrap_or_revert();
+        Urls {
+            dict_uref: *dict_uref
+        }
+    }
+
+    pub fn set(&self, address: &AccountHash, url: &str) {
+        if !is_valid_url(url) {
+            runtime::revert(ContractError::BadUrlFormat);
+        }
+        storage::dictionary_put(self.dict_uref, &address.to_string(), url);
+    }
+    
+    pub fn delete(&self, address: &AccountHash) {
+        storage::dictionary_put(self.dict_uref, &address.to_string(), "");
+    }
+
+    pub fn get(&self, address: &AccountHash) -> String {
+        let url: String = storage::dictionary_get(self.dict_uref, &address.to_string())
+            .unwrap_or_revert()
+            .unwrap_or_revert_with(ContractError::NotFound);
```

Comment:
It looks like on the former out of the two lines above is needed (unless I'm missing something)

Reply:
This one is no longer relevant after @deuszex educated me that dictionary_get returns with Result<Option, E> :)